design-documents/stateless-api-token.md

```diff
@@ -0,0 +1,147 @@
+## Introduce stateless web API tokens
+&nbsp;&nbsp;&nbsp;&nbsp;Authentication tokens used to access web APIs (REST, SOAP, GraphQL) should contain enough
+information to authenticate a user without having to store session data on server side.
+ 
+This approach has following advantages:
+* In case of distributed deployment instances don't have to access single storage/service to authenticate a user
+* Saves storage space
+* Faster to decode/decrypt than load from storage
+* Don't have to implement token protection mechanism on backend to keep tokens from being stolen
+ 
+ 
+### Current situation
+&nbsp;&nbsp;&nbsp;&nbsp;Magento stores tokens in RDMS table named `oauth_token` which is originally meant to store long-term integration tokens
+but also used for admin/customer user tokens. Token rows also contain user type, issued data and expiration data which
+is retrieved when authenticating users for web API via token passed in the header. Admin and customer tokens may be
+manually expired by admin users by updating `oauth_token` table data. Tokens themselves are random strings of static
+length.
+ 
+With described above circumstances in order to perform authentication Magento instances have to have access to same DB
+and perform (relatively) expensive SQL queries to DB. Te more users Magento has the more tokens it has to store
+until they expire and run an extra cron job to remove expired ones.
+ 
+ 
+### Overview of new stateless tokens implementation
```

Comment:
More scenarios should be covered not only user authentication. We have more scenarios that require stateless tokens like 3rd party integrations, webhooks, BFF communication.

Reply:
I think right now all we can do is provide API for encoding and decoding tokens. I'll add such API description to the doc